Neos.Neos/Classes/Controller/Service/NodesController.php

```diff
@@ -110,6 +114,11 @@ public function indexAction(
         string $contextNode = null,
         array|string $nodeIdentifiers = []
     ): void {
+        $searchTerm = $searchTerm === '' ? null : SearchTerm::fulltext($searchTerm);
```

Comment:
Ah, that's probably why we have to add some exceptions to the PHPStan baseline.
My IDE will complain about the inline type change:
![image](https://github.com/user-attachments/assets/1eb89134-fb0a-4b4d-ba7a-095d3bd3d4f9)
And IMO it is correct. We really shouldn't do that

Reply:
btw: if you do use a different variable name, the error is detected by the IDE (or at least by the [Php Inspections plugin](https://plugins.jetbrains.com/plugin/7622-php-inspections-ea-extended-)):

![image](https://github.com/user-attachments/assets/78f0162e-86a8-4099-a733-f8ad61da86b2)

It also suggests a follow-up change that makes sense:

![image](https://github.com/user-attachments/assets/12d80f1a-9e14-4327-b20c-14c06813da31)